lib/engine/step/buy_sell_par_shares.rb

```diff
@@ -51,15 +51,21 @@ def description
       end
 
       def pass_description
-        if @current_actions.empty?
+        if @round.player_actions.empty?
           'Pass (Share)'
         else
           'Done (Share)'
         end
       end
 
       def round_state
-        { players_sold: Hash.new { |h, k| h[k] = {} } }
+        { players_sold: Hash.new { |h, k| h[k] = {} }, player_actions: [] }
```

Comment:
why don't you make the exchange a part of your custom buy_sell_par_shares, that way you don't have to make this global